File: java/abstractClass/abstractClass/src/abstractClass/abstractClass/src/abstractClass.java
// メインクラス
class abstractClass {	// abstractClassクラスの定義

	// Javaのエントリポイント
	public static void main(String[] args) {	// mainメソッドの定義

		// abstractClass3オブジェクトインスタンスをabstractClass2のオブジェクトに格納.
		abstractClass2 aco2_1 = new abstractClass3();	// abstractClass3オブジェクトインスタンスを生成し, abstractClass2オブジェクトaco2_1に格納.

		// aco2_1に値をセット.
		aco2_1.set(10, "ABC");	// setメソッドで10と"ABC"をセット.

		// aco2_1の内容を出力.
		aco2_1.output();	// outputメソッドで整数値と文字列を出力.

		// abstractClass4オブジェクトインスタンスをabstractClass2のオブジェクトに格納.
		abstractClass2 aco2_2 = new abstractClass4();	// abstractClass4オブジェクトインスタンスを生成し, abstractClass2オブジェクトaco2_2に格納.

		// aco2_2に値をセット.
		aco2_2.set(20, "XYZ");	// setメソッドで20と"XYZ"をセット.

		// aco2_2の内容を出力.
		aco2_2.output();	// outputメソッドで整数値と文字列を出力.

	}

}

File: cs/abstract/abstract/src/abstract/abstract/AbstractClass.cs
// 抽象クラス
abstract class AbstractClass    // 抽象クラスAbstractClassの定義
{

    // メソッドの定義
    public AbstractClass()  // コンストラクタ
    {
        // とりあえずメソッド名の出力.
        System.Console.WriteLine("AbstractClass()");    // System.Console.WriteLineでメソッド名を出力.
    }

    public AbstractClass(int i, string str) // コンストラクタ(引数あり)
    {
        // とりあえずメソッド名の出力.
        System.Console.WriteLine("AbstractClass(int i, string str)");    // System.Console.WriteLineでメソッド名を出力.
    }

    virtual public void Set(int i, string str)  // 整数値と文字列をセット.
    {
        // とりあえずメソッド名の出力.
        System.Console.WriteLine("Set(int i, string str)");    // System.Console.WriteLineでメソッド名を出力.
    }

    virtual public void Output()    // 出力.
    {
        // とりあえずメソッド名の出力.
        System.Console.WriteLine("Output()");    // System.Console.WriteLineでメソッド名を出力.
    }

}